### PR TITLE
Wphu/add sex check

### PR DESCRIPTION
**Pull request by williamphu**
More merges:

First one from `get_ibd` is more simple - moving the logger so that remap_hts has values, and updating arguments being passed

and the second is the `sex_check` file that was originally a notebook.

### CODE REVIEW

File: tgg/seqr_scripts/sex_check.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    # will need to generalize more for gnomad methods
+    outdir = dirname(callset)
+    mt_name = callset.split("/")[-1].strip("\.mt")
+    logging.info("Reading matrix table for callset: {callset}".format(**locals()))
+    logging.info("Using chromosome Y coverage? {use_y_cov}".format(**locals()))
+
+    mt = hl.read_matrix_table(callset)
+    
+    # filter to SNVs and biallelics
+    # filter to biallelics might be a function
```
Comment (ch-kr) [inline tgg/seqr_scripts/sex_check.py:168]:
```suggestion
    # Filter to SNVs and biallelics
```

Comment (ch-kr) [reply tgg/seqr_scripts/sex_check.py:168]: I found from `gnomad.utils.annotations import bi_allelic_expr`, but this is fine

File: tgg/seqr_scripts/sex_check.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    mt = mt.filter_rows((hl.len(mt.alleles) == 2) & hl.is_snp(mt.alleles[0], mt.alleles[1]))
+    
+    # filter to pass variants only (empty set)
+    # gnomad methods - will need to make this an optional argument
```
Comment (ch-kr) [inline tgg/seqr_scripts/sex_check.py:172]:
```suggestion
    # Filter to pass variants only (empty set)
    # TODO: Make this an optional argument before moving to gnomad_methods
```

File: tgg/seqr_scripts/sex_check.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    mt = mt.filter_rows(mt.filters.length() == 0, keep=True)
+    
+    #infer build:
```
Comment (ch-kr) [inline tgg/seqr_scripts/sex_check.py:175]:
```suggestion
    # Infer build:
```

File: tgg/seqr_scripts/sex_check.py [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+    build = get_reference_genome(mt.locus).name
+    logging.info("Build inferred as {build}".format(**locals()))
```
Comment (ch-kr) [inline tgg/seqr_scripts/sex_check.py:177]:
```suggestion
    logging.info("Build inferred as %s", build)
```

File: tgg/seqr_scripts/sex_check.py [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    
+    logging.info("Inferring sex...")
+    #for production change female and male to XX and XY
```
Comment (ch-kr) [inline tgg/seqr_scripts/sex_check.py:180]:
```suggestion
    # TODO: Change female and male to XX and XY
```

File: tgg/seqr_scripts/sex_check.py [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+        sex_ht = sex_ht.select(sex_ht.is_female, sex_ht.f_stat, sex_ht.n_called, sex_ht.expected_homs, sex_ht.observed_homs, sex_ht.sex, sex_ht.chrY_mean_cov, sex_ht.chr20_mean_cov, sex_ht.normalized_y_coverage)
+        
+    else:
+        sex_ht = run_hails_impute_sex(mt, build, outdir, mt_name, male_fstat_threshold, female_fstat_threshold, aaf_threshold)
+        sex_ht = sex_ht.annotate(ambiguous_sex=hl.is_missing(sex_ht.is_female))
+        sex_expr = hl.cond(sex_ht.ambiguous_sex, "ambiguous_sex", hl.cond(sex_ht.is_female, "female", "male"))
+        sex_ht = sex_ht.annotate(sex=sex_expr)
+        sex_ht = sex_ht.select(sex_ht.is_female, sex_ht.f_stat, sex_ht.n_called, sex_ht.expected_homs, sex_ht.observed_homs, sex_ht.sex)
+
+    outfile = outdir + f"/sex_{mt_name}.txt"
```
Comment (ch-kr) [inline tgg/seqr_scripts/sex_check.py:205]:
```suggestion
    outfile = f"{outdir}/sex_{mt_name}.txt"
```

File: tgg/seqr_scripts/sex_check.py [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+
+if __name__ == '__main__': 
+    parser = argparse.ArgumentParser(description='This script infers the sex of samples')
+    parser.add_argument('-i', '--callset', required=True, help='Path to Callset MatrixTable')
+    parser.add_argument('-u', '--use-y-cov', help='bool for whether or not to use chrY coverage', action='store_true')
+    parser.add_argument('-y', '--y-cov-threshold', help='chrY coverage threshold', default=0.1)
+    parser.add_argument('-m', '--male-fstat-threshold', help='male_fstat_threshold for hails impute_sex', default=0.75)
+    parser.add_argument('-f', '--female-fstat-threshold', help='female_fstat_threshold for hails impute_sex', default=0.50)
+    parser.add_argument('-a', '--aaf-threshold', help='aaf_threshold for hails impute_sex', default=0.05)
+    parser.add_argument('-c', '--call-rate-threshold', help='call_rate_threshold required to use chrY variant', default=0.25)
```
Comment (ch-kr) [inline tgg/seqr_scripts/sex_check.py:220]:
```suggestion
    parser = argparse.ArgumentParser(description='This script infers the sex of samples')
    parser.add_argument('-i', '--callset', required=True, help='Path to Callset MatrixTable')
    parser.add_argument('-u', '--use-y-cov', help='Boolean for whether or not to use chrY coverage', action='store_true')
    parser.add_argument('-y', '--y-cov-threshold', help='chrY coverage threshold', default=0.1)
    parser.add_argument('-m', '--male-fstat-threshold', help='Male f-stat threshold for `hl.impute_sex`', default=0.75)
    parser.add_argument('-f', '--female-fstat-threshold', help='Female f-stat threshold for `hl.impute_sex`', default=0.50)
    parser.add_argument('-a', '--aaf-threshold', help='Alternate allele frequency threshold for `hl.impute_sex`', default=0.05)
    parser.add_argument('-c', '--call-rate-threshold', help='Call rate threshold used to filter chrY variants', default=0.25)
```

File: tgg/seqr_scripts/sex_check.py [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+    
+    args = parser.parse_args()
+    main(args)
```
Comment (ch-kr) [inline tgg/seqr_scripts/sex_check.py:223]: might also be helpful to add slack configuration (also could be a TODO)

**Review by ch-kr — CHANGES_REQUESTED** [review]
thanks for adding the sex check code! I have some suggestions but wasn't sure how much to add, as I'm hoping the summer students can productionize this code.

Other small comments/questions:

- The imports in the IBD script should likely be reordered
- The IBD script includes the `no_whole_stage_codegen` flag -- is this still required?
- The code has some formatting issues (easily fixed with BLACK)

File: tgg/seqr_scripts/sex_check.py [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+    else:
+        chr20 = "20"
+        chrY = "Y"
+
+    logging.info("Filtering to chromosome 20 and non-par regions on chromosome Y to calculate normalized Y coverage...")
+    sex_mt = hl.filter_intervals(mt, [hl.parse_locus_interval(chr20, reference_genome=build), hl.parse_locus_interval(chrY, reference_genome=build)])
+    sex_mt = sex_mt.filter_rows((sex_mt.locus.contig == chr20) | (sex_mt.locus.in_y_nonpar()), keep = True)
+
+
+    # filter to common snvs above defined callrate (because only biallelic, should now only have one index in the array)
```
Comment (klaricch) [inline tgg/seqr_scripts/sex_check.py:36]: here and throughout snvs -> SNVs

**Pull request merged**